src/test/java/seedu/duke/commands/ShowRateCommandTest.java

```diff
@@ -22,4 +29,62 @@ public void testExecute_correctSyntax_shouldNotException() {
             fail();
         }
     }
+
+    /**
+     * Potential exceptions are all caught and handled by execute, so if
+     * no exception is thrown in these cases, then it means the code
+     * does not break under any of these circumstances.
+     */
+    @Test
+    public void testExecute_incorrectSyntax_shouldHandleThrowInvalidShowRateArgumentException() {
+        try {
+            Ui ui = new Ui();
+            AccountList accounts = new AccountList();
+            ShowRateCommand cmd1 = new ShowRateCommand("show-rate THB");
+            ShowRateCommand cmd2 = new ShowRateCommand("show-rate THB SGD USD MYR");
+            cmd1.execute(ui, accounts);
+            cmd2.execute(ui, accounts);
+        } catch (Exception e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void testExecute_invalidCurrencyOrNumber_shouldHandleIllegalArgumentException() {
+        try {
+            Ui ui = new Ui();
+            AccountList accounts = new AccountList();
+            ShowRateCommand cmd1 = new ShowRateCommand("show-rate THB XYZ");
+            ShowRateCommand cmd2 = new ShowRateCommand("show-rate THB SGD CS_2113");
+            cmd1.execute(ui, accounts);
+            cmd2.execute(ui, accounts);
+        } catch (Exception e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void testExecute_negativeNumber_shouldHandleInvalidNumberException() {
+        try {
+            Ui ui = new Ui();
+            AccountList accounts = new AccountList();
+            ShowRateCommand cmd = new ShowRateCommand("show-rate THB SGD -1.24");
+            cmd.execute(ui, accounts);
+        } catch (Exception e) {
+            fail();
+        }
+    }
+
+    public void testGetRateString_negativeNumber_shouldThrowInvalidNumberException() {
+        try {
+            Ui ui = new Ui();
```

Comment:
Did you use the the UI instance somewhere?

Reply:
Good catch, I am going to make another PR changing the testing for show-rate, so I can remove it then.